packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -0,0 +1,276 @@
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import Component from 'base-component'
+import decorate from 'apply-decorators'
+import PropTypes from 'prop-types'
+import React from 'react'
+import Wizard, { Section } from 'wizard'
+import { map } from 'lodash'
+import { Toggle } from 'form'
+import { injectIntl } from 'react-intl'
+import { linkState } from 'reaclette-utils'
+import { Container, Row, Col } from 'grid'
+import { provideState, injectState } from 'reaclette'
+import { SelectPool, SelectPif } from 'select-objects'
+import { addSubscriptions, connectStore } from 'utils'
+import {
+  getBondModes,
+  createBondedNetwork,
+  createNetwork,
+  subscribeCurrentUser,
+} from 'xo'
+import {
+  createSelector,
+  createGetObject,
+  getIsPoolAdmin,
+  getUser,
+} from 'selectors'
+
+import styles from './index.css'
+import Page from '../../page'
+
+const getInitialState = () => ({
+  bondedNetwork: false,
+  bondMode: undefined,
+  description: '',
+  mtu: '',
+  name: '',
+  pif: undefined,
+  pifs: undefined,
+  vlan: '',
+})
+
+const NewNetwork = decorate([
+  addSubscriptions({
+    user: subscribeCurrentUser,
+  }),
+  connectStore(() => ({
+    isAdmin: createSelector(
+      getUser,
+      user => user && user.permission === 'admin'
+    ),
+    isPoolAdmin: getIsPoolAdmin,
+    pool: createGetObject((_, props) => props.location.query.pool),
+  })),
+  injectIntl,
+  provideState({
+    initialState: getInitialState,
+    effects: {
+      linkState,
+      onChangeBondedNetwork: () => ({ bondedNetwork }) => ({
+        ...getInitialState(),
+        bondedNetwork: !bondedNetwork,
+      }),
+      onChangePif: (_, value) => ({ bondedNetwork }) =>
+        bondedNetwork ? { pifs: value } : { pif: value },
+      reset: getInitialState,
+    },
+    computed: {
+      pifPredicate: (_, { pool }) => pif =>
+        pif.vlan === -1 && pif.$host === (pool && pool.master),
+    },
+  }),
+  injectState,
+  class extends Component {
+    static contextTypes = {
+      router: PropTypes.object,
+    }
+
+    componentWillMount() {
+      getBondModes().then(bondModes => this.setState({ bondModes }))
+    }
+
+    componentWillReceiveProps(props) {
+      this.forceUpdate()
+    }
+
+    _create = () => {
+      const { bondModes } = this.state
+      const { pool, state } = this.props
+      const {
+        bondedNetwork,
+        bondMode,
+        description,
+        mtu,
+        name,
+        pif,
+        pifs,
+        vlan,
+      } = state
+      return bondedNetwork
+        ? createBondedNetwork({
+            bondMode: bondMode || bondModes[0],
```

Comment:
If a bond mode is mandatory, then the user shouldn't be able to fill the form without one.